packages/zoe/src/contracts/loan/updateDebt.js

```diff
@@ -67,41 +70,16 @@ export const makeDebtCalculator = debtCalculatorConfig => {
     }
   };
 
-  const addToDebtWhenNotified = lastCount => {
-    const processUpdate = newState => {
-      const { updateCount, value } = newState;
-      if (updateCount === undefined) {
-        const reason = 'PeriodNotifier should not publish a final value';
-        debtNotifierUpdater.fail(reason);
-        throw Error(reason);
-      }
-
+  const handleDebt = async () => {
+    for await (const value of makeAsyncIterableFromNotifier(periodNotifier)) {
```

Comment:
We really need to rename this helper method.

Reply:
See #2513

Reply:
No longer appears. Resolving.